ARCs/arc-0060.md

```diff
@@ -0,0 +1,330 @@
+
+#### **Interface `SignDataFunction`**
+
+A wallet function `signData` is defined by the interface:
+
+```tsx
+export type SignDataFunction = {
+    data: StdData,
+    metadata: StdSignMetadata,
+    signer: Ed25519Pk,
+}
+    => Promise<(SignedDataStr | null)>;
+```
+
+- `data` is a `StdData` object representing the structured data being signed (defined below).
+- `metadata` is a `StdSignMetadata` objects (defined below) that provide additional information on the data being signed, such as the `data` encoding and the JSON schema.
+- `signer` is a `Ed25519Pk` object (defined below) that represents the signer public key.
+
+The `signData` function returns a `SignedDataStr` object or, in case of error, rejects the promise with an error object `SignDataError`.
+
+#### Interface `StdData`
+
+The `StdData` object is a string that comply with the `metadata` JSON schema. It **MUST** be the byte representation of a canonicalized JSON object, following the <a href="https://www.rfc-editor.org/rfc/rfc8785">RFC 8785</a>.
+
+```tsx
+export type StdData = string;
+```
+
+The `StdData` must be validated with respect the JSON schema provided with `metadata` (defined below).
+
+#### Interface `Ed25519Pk`
+
+An `Ed25519Pk` object is a 32-byte public key, point of the ed25519 elliptic curve. The key **MUST NOT** be transformed into an Algorand address.
+
+```tsx
+export type Ed25519Pk = Uint8Array;
+```
+
+> The wallet **MAY** want to operate with standard Algorand address directly. The transformation from a generic key to an Algorand address is left to the implementation. See <a href="https://developer.algorand.org/docs/get-details/accounts/#transformation-public-key-to-algorand-address">Public Key to Algorand Address</a> section of the developer portal.
+
+#### Interface `SignedDataStr`
+
+`SignedDataStr` is the produced 64-byte array, ed25519 digital signature, of the signed data.
+
+```tsx
+export type SignedDataStr = Uint8Array;
+```
+
+#### Enum `ScopeType`
+
+The `ScopeType` enumerates constant strings with the scope of a signing action.
+
+This ARC introduces two scope types.
+
+| ScopeType | Description |
+| --- | --- |
+| MSGSIG | Signature of a simple message. This is the most generic scope. |
+| LSIG | Signature of an Algorand program for delegation. |
+| ... | ... |
+
+Any extension of this ARC **SHOULD** adopt the `ScopeType` above, or introduce a new scope.
+
+#### Interface `StdSignMetadata`
+
+A `StdSignMetadata` object specifies metadata of a `StdSignData` that is being signed.
+
+```tsx
+export interface StdSignMetadata {
+    /**
+    * The scope value of the signing data request.
+    */
+    scope: ScopeType;
+
+    /**
+    * Canonical representation of the JSON schema for the signing data.
+    */
+    schema: string;
+
+    /**
+    * Optional message explaining the reason for the signature.
+    */
+    message?: string;
+
+    /**
+    * Optional encoding used to represent the signing data.
+    */
+    encoding?: string;
```

Comment:
What are possible encodings that should be supported?

Reply:
+1 to clarifying this

Reply:
It can be any encoding. If the wallet does not support it, it will just reject the call. I make it explicit here https://github.com/algorandfoundation/ARCs/pull/284/commits/88923d3723674870d1f442e643f02834a99afb3e